bpf/lib/egress_policies.h

```diff
@@ -87,10 +88,7 @@ bool egress_gw_reply_needs_redirect(struct iphdr *ip4, __u32 *tunnel_endpoint,
 	if (!egress_policy)
 		return false;
 
-	/* If the gateway IP is 0.0.0.0 it means this is an excluded CIDR, so
-	 * skip reply redirect
-	 */
-	if (!egress_policy->gateway_ip)
+	if (egress_gw_policy->gateway_ip == EGRESS_GATEWAY_EXCLUDED_CIDR)
```

Comment:
I think we still need to handle the `EGRESS_GATEWAY_NO_GATEWAY` case (ditto for `egress_gw_snat_needed`).

Also `s/egress_gw_policy/egress_policy/`

Reply:
Hmm I'm not sure the no gateway case need to be handled here :thinking:. Those two functions are on the "gateway node" so it means that we would encounter this case whenever loosing the gateway node but traffic still get to it for some reasons. So I don't think it matters that much to snat or not in `egress_gw_reply_needs_redirect` but perhaps we could also drop the traffic in `egress_gw_reply_needs_redirect` whenever this happen (in stage 2 of the change then). WDYT?

And indeed bad copy paste for the `egress_gw_policy` thing :man_facepalming:.

Reply:
Ah and also doing something special here could maybe be racy? Like if go from no gateway -> some gateway is set, and if the agent wherever the pod is scheduled get the information first... It shouldn't be a huge issue though as it will likely go away very soon and if it was previously in a bad state connection was broken anyway at some point so it simply add a very small delay.

Reply:
Well actually you were totally right sorry, I thought this was somehow executed only on the gateway node but it is done also on the "client node" as well. And yeah not handling this mean that traffic will get snated to 0.0.0.0 :sweat:... It should works now. In the last test run there was something about the egress policy entry not being created on the new "no gateway test" in `test/k8s/egress.go` but I cannot reproduce locally :thinking:...